README.md

```diff
@@ -33,26 +33,56 @@ For [conda](https://docs.conda.io/) and [spack](https://spack.io/) installation
 
 ## Documentation
 
-You can find the documentation for torchgeo on [ReadTheDocs](https://torchgeo.readthedocs.io).
+You can find the documentation for TorchGeo on [ReadTheDocs](https://torchgeo.readthedocs.io).
 
-## Example usage
+## Example Usage
 
-The following sections give basic examples of what you can do with torchgeo. For more examples, check out our [tutorials](https://torchgeo.readthedocs.io/en/latest/tutorials/getting_started.html).
+The following sections give basic examples of what you can do with TorchGeo. For more examples, check out our [tutorials](https://torchgeo.readthedocs.io/en/latest/tutorials/getting_started.html).
 
-### Train and test models using our PyTorch Lightning based training script
+First we'll import various classes and functions used in the following sections:
 
-We provide a script, `train.py` for training models using a subset of the datasets. We do this with the PyTorch Lightning `LightningModule`s and `LightningDataModule`s implemented under the `torchgeo.trainers` namespace.
-The `train.py` script is configurable via the command line and/or via YAML configuration files. See the [conf/](conf/) directory for example configuration files that can be customized for different training runs.
+```python
+from torch.utils.data import DataLoader
+from torchgeo.datasets import CDL, Landsat7, Landsat8, TropicalCycloneWindEstimation, stack_samples
+from torchgeo.samplers import RandomGeoSampler
+```
 
-```console
-$ python train.py config_file=conf/landcoverai.yaml
+### Composing geospatial datasets
+
+Many use cases involve working with geospatial data and combining it in intelligent ways. In this example, we assume that the user has Landsat 7 and 8 imagery downloaded. We first create a single dataset containing all Landsat imagery. We only use the Landsat 7 bands for Landsat 8 and take the union between both datasets.
+
+```python
+landsat7 = Landsat7(root="...")
+landsat8 = Landsat8(root="...", bands=landsat7.bands)
+landsat = landsat7 | landsat8
+```
+
+Next, we take the intersection between this dataset and the Cropland Data Layer (CDL) dataset. We want to take the intersection instead of the union to ensure that we only sample from regions that have both Landsat and CDL data. Note that we can automatically download and checksum CDL data. Also note that each of these datasets may contain files in different coordinate reference systems (CRS) or resolutions, but TorchGeo automatically ensures that a matching CRS and resolution is used.
+
+```python
+cdl = CDL(root="...", download=True, checksum=True)
+dataset = landsat & cdl
```

Comment:
A couple of questions:
- Have you tested this with Landsat7/8 dataset?
- Since you are using both 7 and 8 with CDL data, I think people will wonder how time is handled. Should we explain that? How is it handled here?

Reply:
> Have you tested this with Landsat7/8 dataset?

Nope.

> Since you are using both 7 and 8 with CDL data, I think people will wonder how time is handled. Should we explain that? How is it handled here?

It isn't. If you check our current sampler implementations, we more or less ignore time. We choose a random tile, and a random patch within that tile, and the returned bounding box has its mint/maxt set to that of the tile. We don't yet have any time-aware sampler implementations.